app/models/institutions_archive.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+require_dependency 'institution'
+
+class InstitutionsArchive < Institution
+  self.table_name = 'institutions_archives'
+
+  # class methods
+  def self.archive_previous_versions
+    production_version = Version.current_production.number
+    previous_version = Version.previous_production.number
+
+    begin
+      ActiveRecord::Base.transaction do
+        create_archives(previous_version, production_version)
+        Institution.where('version >= ? and version < ?', previous_version, production_version).delete_all
+      end
+    rescue ActiveRecord::StatementInvalid => exception
+      notice = 'There was an error occurring at the database level'
+      error_msg = exception.original_exception.result.error_message
+      process_exception(notice, error_msg, production_version, previous_version)
+    rescue StandardError => exception
+      notice = 'There was an error of unexpected origin'
+      error_msg = exception.message
+      process_exception(notice, error_msg, production_version, previous_version)
+    end
+  end
+
+  def self.create_archives(previous_version, production_version)
+    str = <<-SQL
+      INSERT INTO institutions_archives
+        SELECT *
+          FROM institutions
+          WHERE version >= ? and version < ? ;
```

Comment:
I'm confused about how this might archive anything other than `previous_version`

Reply:
... but now I understand